File: integration-test/java/spring-app-r2dbc/src/main/java/com/qualityminds/lazyval/integration/domain/OrderRepository.java
package com.qualityminds.lazyval.integration.domain;

import com.qualityminds.lazyval.integration.shared.Isbn;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

import java.util.UUID;

/**
 * Reactive counterpart to the blocking port in the {@code spring-app} scenario. The whole point of
 * this module is that nothing blocks between the HTTP boundary and the driver, so the port itself is
 * reactive rather than hiding a {@code block()} in the adapter.
 */
public interface OrderRepository {

    Mono<Order> save(Order order);

    Flux<Order> findAll();

    Mono<Order> getById(UUID id);

    Flux<Order> findByIsbn(Isbn isbn);
}

File: integration-test/kotlin/spring-app-r2dbc/src/main/kotlin/com/qualityminds/lazyval/integration/domain/OrderRepository.kt
package com.qualityminds.lazyval.integration.domain

import com.qualityminds.lazyval.integration.shared.Isbn
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono
import java.util.*

/**
 * Reactive counterpart to the blocking port in the kotlin/spring-app scenario. Reactor types rather
 * than coroutines: R2DBC is reactive by construction, and this scenario deliberately keeps the
 * coroutine adapter out of the path so a startup failure points at Spring's dependency resolution
 * of the KSP-generated converter config rather than at the bridge.
 */
interface OrderRepository {

    fun save(order: Order): Mono<Order>

    fun findAll(): Flux<Order>

    fun getById(id: UUID): Mono<Order>

    fun findByIsbn(isbn: Isbn): Flux<Order>
}
